base/error.go

```diff
@@ -237,6 +237,32 @@ func IsDocNotFoundError(err error) bool {
 	}
 }
 
+func IsTemporaryKvError(err error) bool {
+	if err == nil {
+		return false
+	}
+	if errors.Is(err, ErrTimeout) {
+		return true
+	}
+	if errors.Is(err, gocb.ErrAmbiguousTimeout) {
+		return true
+	}
+	if errors.Is(err, gocb.ErrUnambiguousTimeout) {
+		return true
+	}
+	if errors.Is(err, gocb.ErrOverload) {
+		return true
+	}
+	if errors.Is(err, gocb.ErrTemporaryFailure) {
+		return true
+	}
+	if errors.Is(err, gocb.ErrCircuitBreakerOpen) {
+		return true
+	}
```

Comment:
nit style, maybe this is easier to read because you know these will all flag as true:

```suggestion
	for _, tempKVErr := range []error{ErrTimeout, gocb.ErrAmbiguousTimeout, gocb.ErrUnambiguousTimeout, gocb.ErrOverload, gocb.ErrTemporaryFailure, gocb.ErrCircuitBreakerOpen}) {
	    if errors.Is(err, tempKVErr) {
			return true
		}
	}
```